src/Raven.Server/Documents/Sharding/Handlers/Batches/ShardedBatchCommand.cs

```diff
@@ -79,18 +79,28 @@ public IEnumerator<SingleShardedCommand> GetEnumerator()
                 continue;
             }
 
-            var shard = _databaseContext.GetShardNumber(_context, cmd.Id);
-            var commandStream = bufferedCommand.CommandStream;
-            var stream = cmd.Type == CommandType.AttachmentPUT ? AttachmentStreams[streamPosition++] : null;
+            int shardNumber;
+            if (bufferedCommand.IsServerSideIdentity)
+            {
+                var bucket = ShardHelper.GetRandomBucket(); // we do not care about bucket here, just need a random number with max value
```

Comment:
That isn't actually a bucket id. It is just something that we hash.
We are probably better generating a alphanumeric value here, instead.
So generate a random value, then use the Base26 encoding to get a random looking value.

A numeric value may lead us to assume this _is_ the bucket.

Reply:
I agree. 
If my math is correct we will need only 5 chars in Base26. (given that the distribution is uniform)